website/blog/2024-04-04-angular-upgrade-top-5-tips.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "Angular Upgrade Journey: Top 5 things to keep in mind"
+author: Sairama Krishna Bonala
+---
+
+Taking up an Angular upgrade can be both exciting and challenging. As technology evolves, it's essential to keep up to date for better performance and security. This guide explains key considerations and best practices to make the Angular upgrade journey smoother and successful.
+
+<!--truncate-->
+
+## 1. Check Compatibility
+- Check [Support timelines](https://angular.io/guide/releases#support-window) for the Active and LTS versions.
+- Consider upgrading Angular in incremental versions especially if the project is moving from a significantly older version. 
+- Angular provides comprehensive documentation on breaking changes between versions. Utilize tools like the [Angular Update Guide](https://update.angular.io/), which guides the necessary changes. Look specifically for any breaking changes or deprecated features.
+- Consider creating a separate branch for the upgrade process to avoid disrupting the main development workflow until the upgrade is complete.
+
+:::tip
+Update Node version as mentioned in the [Angular version compatibility document](https://angular.io/guide/versions). 
+:::
+
+## 2. Use Angular CLI
+- Angular CLI simplifies various tasks in the upgrade process, automates many tedious tasks, and ensures best practices are followed.
+- Use the command `ng update` to automatically update the project's dependencies and modify configuration files to match the latest Angular version.
+- Be vigilant in checking for any breaking changes and compilation issues that may arise during the upgrade process. Address these issues promptly to maintain the integrity of the application.
+
+## 3. Update Dependencies
+- In addition to upgrading Angular itself, don't forget to update other project dependencies.
+- Check the release notes of Angular and other dependencies used in the project to identify compatibility issues or breaking changes.
+- Incrementally update each dependency version and thoroughly test the application after each update to catch and resolve any issues early on.
+- Check for the changes in Git/npm repositories that may impact the project and adjust accordingly to avoid unexpected issues.
+
+:::info
+Consider supporting dependencies when upgrading to Angular Ivy from View Engine. Projects on Angular 16 and later no longer support View Engine libraries.
```

Comment:
Shouldn't we specify users to be ready with all the dependencies by the proposed date?